lua/oil/config.lua

```diff
@@ -89,6 +89,7 @@ local default_config = {
   },
   -- Configuration for the floating window in oil.open_float
   float = {
+    open_by_default = false,
```

Comment:
The `float` table contains values to customize the display of the floating window. I think it would make more sense to put this at the top level and call it something like `default_floating_win`